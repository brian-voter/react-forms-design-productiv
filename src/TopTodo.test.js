import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TopTodo from "./TopTodo.js";
import { initialTodos } from "./_testCommon.js";

describe("TopTodo Component smoke and snapshot", function () {
  test("smoke test - renders without crashing", function () {
    render(<TopTodo todos={initialTodos} />);
  });

  test("snapshot test", function () {
    const { container } = render(
      <TopTodo
        todos={initialTodos} />);
    expect(container).toMatchSnapshot();
  });
});

describe("TopTodo shows correct todo", function () {
  test("displays highest priority todo", function () {
    const { queryByText } = render(<TopTodo todos={initialTodos} />);

    expect(queryByText("Make dinner")).toBeInTheDocument();
    expect(queryByText("Cook something healthy")).toBeInTheDocument();
  });
});