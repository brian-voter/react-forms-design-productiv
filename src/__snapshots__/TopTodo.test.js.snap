// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo Component smoke and snapshot snapshot test 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Make dinner
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Cook something healthy
      </small>
    </div>
  </div>
</div>
`;
